File: src/main/java/io/vertigo/shiny/components/data/json/ShinyJson.java
package io.vertigo.shiny.components.data.json;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyJson(
		String title,
		String json) implements ShinyComponent {

	@Override
	public String type() {
		return "json";
	}


	public ShinyJson {
		Assertion.check().isNotBlank(json, "JSON string cannot be blank");
	}
}

File: banshee/src/models/data/json/ShinyJson.ts
import { ShinyComponent } from '../../../ShinyComponent';

export class ShinyJson implements ShinyComponent {
  title: string;
  json: string;
  type: string = 'json';

  constructor(title: string, json: string) {
    this.title = title;
    this.json = json;
  }
}
